tests/functional/object/mcg/test_bucket_notifications.py

```diff
@@ -144,3 +155,360 @@ def test_bucket_notifications(
+            )
+            buckets_to_topics[bucket] = topic
+
+        # 3. Write some objects to each bucket
+        for bucket in buckets_to_topics.keys():
+            file_dir = os.path.join(test_directory_setup.origin_dir, bucket)
+            objs = write_random_test_objects_to_bucket(
+                io_pod=awscli_pod,
+                bucket_to_write=bucket,
+                file_dir=file_dir,
+                amount=5,
+                pattern=f"{bucket}-",
+                mcg_obj=mcg_obj,
+            )
+            buckets_to_written_objs[bucket] = set(objs)
+
+        # 4. Verify that the expected events were received by Kafka
+        def _get_events_by_topic():
+            events_by_topic = dict()
+            for topic in buckets_to_topics.values():
+                parsed_events_set = set(
+                    event["s3"]["object"]["key"]
+                    for event in notif_manager.get_events(topic)
+                )
+                events_by_topic[topic] = parsed_events_set
+            return events_by_topic
+
+        buckets_to_received_events = dict()
+        accumulated_delta = set()
+        try:
+            for events in TimeoutSampler(
+                timeout=300, sleep=10, func=_get_events_by_topic
+            ):
+                accumulated_delta.clear()  # Clear deltas from the previous iteration
+
+                for bucket, expected_objs_set in buckets_to_written_objs.items():
+                    received_objs_set = events[buckets_to_topics[bucket]]
+                    buckets_to_received_events[bucket] = received_objs_set
+                    accumulated_delta.update(expected_objs_set - received_objs_set)
+                if any(accumulated_delta):
+                    logger.warning(
+                        f"Some expected events were not received: {accumulated_delta}"
+                    )
+                else:
+                    logger.info("Every topic received all its expected events")
+                    break
+        except TimeoutExpiredError as e:
+            raise TimeoutExpiredError(
+                e,
+                f"Some expected events were not received by Kafka: {accumulated_delta}",
+            )
+
+        # 5. Check that no topic received events that it shouldn't have
+        for set_a, set_b in combinations(buckets_to_received_events.values(), 2):
+            assert (
+                not set_a & set_b
+            ), f"Overlap detected: {set_a & set_b} appeared in multiple topics."
+
+    @tier2
+    @polarion_id("OCS-6333")
+    def test_bucket_notifs_shared_topic(
+        self,
+        mcg_obj,
+        awscli_pod,
+        bucket_factory,
+        test_directory_setup,
+        notif_manager,
+    ):
+        """
+        Test setting multiple buckets to send notifications to the same Kafka topic
+
+        1. Enable bucket notifications on the NooBaa CR
+        2. Setup two buckets with bucket notifications to the same topic
```

Comment:
Code is setting up 3 buckets. Please change doc string accordingly

Reply:
Thanks for noticing - fixed in https://github.com/red-hat-storage/ocs-ci/pull/11260/commits/8bf12b8cf5205cabdb5b2d539cd39bd55e034f9b